### ISSUE TITLE
Add videos to link embed for third party platforms

### ISSUE THREAD

**Issue by Tanza3D** [bluesky-social/social-app#5267]
**Is your feature request related to a problem? Please describe.**

Linking posts with videos to other people on third party platforms e.g. Discord or Telegram does not show any video.
![image](https://github.com/user-attachments/assets/d8a25221-048b-4ca2-a937-7c5f5edd9ca4)

**Describe the solution you'd like**

Should show the video video similar to this url using the app's built in video player; https://tanza.me/int/test2.html (visiting this in browser will never load, it only has meta tags)
![image](https://github.com/user-attachments/assets/e450d11c-b4bc-445a-931b-0b3c63d15787)
![image](https://github.com/user-attachments/assets/25feadb3-c568-4816-9e88-1c4750be60ed)

apologies if this is already reported or fixed, i looked around PRs and issues and couldn't find anything

**Comment by IsaacMarovitz** [comment]
Okay, I've dug through how the player works and at least to my understanding, this isn't possible with how the CDN and video player is currently set up. The `og:video:url` meta tags will only work with a proper continuous mp4, but Bluesky splits videos into several MPEG TS files, and then plays them back using a `m3u8` playlist manifest. 

In the case of [this](https://bsky.app/profile/isaacmarovitz.bsky.social/post/3l3zuuy4nai2h) example video I've been testing with, there is an initial parent manifest with multiple resolutions available 

<img width="674" alt="Screenshot 2024-09-14 at 13 18 37" src="https://github.com/user-attachments/assets/0432420b-3c2b-404a-861c-dc3cc1159bfd">

Then a following manifest with each portion of the video
<img width="382" alt="Screenshot 2024-09-14 at 13 18 24" src="https://github.com/user-attachments/assets/4812dd9b-196e-4e28-ac16-5240b36016df">

TLDR; I don't think this is compatible with Open Graph in it's current state unless there is a full video stored somewhere else on the CDN I don't know about.

**Comment by IsaacMarovitz** [comment]
Okay well it turns out the original *is* stored somewhere and can be found using [this](https://github.com/goeo-/cbsky.app). Not sure how to get this through the official API yet.

**Comment by MetaflameDragon** [comment]
The original video is stored in full on each PDS, you can find it by getting a user's PDS (this will most likely be a Bluesky PDS from "the mycosphere", but there are a few self-hosted ones too now). In fact, PDSes store the video in 1080p 3kbps, while the CDN serves it in only up to 720p (for now).

That's how cbsky.app does it, it just links to the raw blob directly. However, I'm not sure if that's the intended way that Bluesky would want to do it - it would fail to load the video if the PDS goes offline (while the video might remain cached on the CDN, same as how AppViews keep posts visible even when a PDS is offline). Image post embeds link to the CDN, so it would probably need more discussion to see if linking to the video blobs directly is fine.

I've been looking into patching the feature in yesterday, first trying to get it working locally.

**Comment by MetaflameDragon** [comment]
To add: the format to retrieve the video blob is:

```
<PDS domain>/xrpc/com.atproto.sync.getBlob?did=<DID>&cid=<Video blob CID>
```

So, for example:

https://bsky.app/profile/pfrazee.com/post/3l3vrskfzqu2s

https://morel.us-east.host.bsky.network/xrpc/com.atproto.sync.getBlob?did=did:plc:ragtjsm2j2vknwkz3zp4oxrd&cid=bafkreif3dm4pfrneqtvnvsy743qfne6uxrfa6w24zevmnv6vvqnhstvg4m

The video CID is returned in the embed field of a post (e.g. via app.bsky.feed.getPostThread).

Also, while the playlist isn't useful for us, the CDN-provided thumbnail is still usable, as it's a simple jpg. (This one is included directly in the embed field too.)

**Comment by icehaku** [comment]
If the video player is too complex for the moment, we could at least get that JPG thumb as a solution for now.
Its quite boring to share a bluesky link video, and it only shows the text part of it: 
https://bsky.app/profile/aisse.bsky.social/post/3l4gscjim6o25
![image](https://github.com/user-attachments/assets/04cbd8a4-0002-4a0c-8614-21c23f558e10)

**Comment by mruac** [comment]
> To add: the format to retrieve the video blob is:
> 
> ```
> <PDS domain>/xrpc/com.atproto.sync.getBlob?did=<DID>&cid=<Video blob CID>
> ```
> 
> So, for example:
> 
> https://bsky.app/profile/pfrazee.com/post/3l3vrskfzqu2s
> 
> https://morel.us-east.host.bsky.network/xrpc/com.atproto.sync.getBlob?did=did:plc:ragtjsm2j2vknwkz3zp4oxrd&cid=bafkreif3dm4pfrneqtvnvsy743qfne6uxrfa6w24zevmnv6vvqnhstvg4m
> 
> The video CID is returned in the embed field of a post (e.g. via app.bsky.feed.getPostThread).
> 
> Also, while the playlist isn't useful for us, the CDN-provided thumbnail is still usable, as it's a simple jpg. (This one is included directly in the embed field too.)

Using the `bsky.social` domain will redirect you to its PDS for the video.
https://bsky.social/xrpc/com.atproto.sync.getBlob?did=did:plc:ragtjsm2j2vknwkz3zp4oxrd&cid=bafkreif3dm4pfrneqtvnvsy743qfne6uxrfa6w24zevmnv6vvqnhstvg4m
